src/client/renderingengine.cpp

```diff
@@ -332,8 +332,20 @@ void RenderingEngine::draw_load_screen(const std::wstring &text,
 		driver->beginScene(true, true, video::SColor(255, 0, 0, 0));
 	}
 
+	int percent_min = 0;
+	int percent_max = percent;
+	if (indefinite) {
+		static float indef_pos = 0;
+		indef_pos += 5 * (dtime / 0.1);
+		if (indef_pos >= 120) {
+			indef_pos = 0;
+		}
```

Comment:
```suggestion
		thread_local float indef_pos = 0;
		indef_pos = (indef_pos + dtime * 0.5f) % 120.0f;
```

Reply:
While I see what you tried to do here, it does not compile. I'll alter my code to something similar though